plugins/flutter_aepedgeidentity/lib/src/aepedgeidentity_identity_map.dart

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2022 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import 'package:flutter_aepedgeidentity/src/aepedgeidentity_identity_item.dart';
+
+/// identity map containing a set of end user identities, keyed on either namespace integration code or the namespace ID of the identity.
+
+class IdentityMap {
+  ///add an `IdentityItem` to this `IdentityMap`
+  Map<String, List<IdentityItem>> identityMap = {};
+
+  void addItem(IdentityItem item, String namespace) {
+    if (item.id.isEmpty) {
+      return;
+    }
+
+    ///add item to the existing namespace
+    IdentityItem itemCopy = copyItem(item);
+
+    if (this.identityMap[namespace] != null) {
+      int index = this
+          .identityMap[namespace]!
+          .indexWhere((element) => equalIds(element.id, itemCopy.id));
+      if (index != -1) {
+        this.identityMap[namespace]![index] = itemCopy;
+      } else
+        [this.identityMap[namespace]!.add(itemCopy)];
+    } else {
+      this.identityMap[namespace] = [itemCopy];
+    }
+  }
+
+  /// Checks if this `IdentityMap` is empty
+  bool isEmpty() {
+    return this.identityMap.keys.toList().isEmpty;
+  }
+
+  /// Gets a list of all namespaces available in this `IdentityMap`
+  List<dynamic> getNamespaces() {
+    return this.identityMap.keys.toList();
+  }
+
+  /// Retrieves the IdentityItems for a given namespace
+  List<IdentityItem>? getIdentityItemsForNamespace(String namespace) {
+    return this.identityMap[namespace];
```

Comment:
here we should return a deep copy of this list

Reply:
L58 List.from(...) creates a shallow copy, but we need a deep copy that copies each item - here are some examples https://stackoverflow.com/questions/51352622/how-to-clone-copy-values-a-complex-object-in-dart-2/51356837#51356837

Reply:
Updated.